core/src/main/java/com/microsoft/applicationinsights/channel/concrete/inprocess/ThrottlingHandler.java

```diff
@@ -0,0 +1,95 @@
+package com.microsoft.applicationinsights.channel.concrete.inprocess;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.TimeZone;
+
+import org.apache.http.Header;
+
+import com.google.common.base.Strings;
+import com.microsoft.applicationinsights.internal.channel.TransmissionHandler;
+import com.microsoft.applicationinsights.internal.channel.TransmissionHandlerArgs;
+import com.microsoft.applicationinsights.internal.channel.common.TransmissionPolicy;
+import com.microsoft.applicationinsights.internal.channel.common.TransmissionPolicyManager;
+import com.microsoft.applicationinsights.internal.logger.InternalLogger;
+
+
+public class ThrottlingHandler implements TransmissionHandler {
+
+	private TransmissionPolicyManager transmissionPolicyManager;
+	private final static String RESPONSE_RETRY_AFTER_DATE_FORMAT = "E, dd MMM yyyy HH:mm:ss";
+	
+	public ThrottlingHandler(TransmissionPolicyManager policy)
+	{
+		this.transmissionPolicyManager = policy;
+	}
+	
+	@Override
+	public void onTransmissionSent(TransmissionHandlerArgs args) {
+		validateTransmissionAndSend(args);
+	}
+
+	public boolean validateTransmissionAndSend(TransmissionHandlerArgs args) {
+		if (args.getRetryHeader() != null && args.getTransmission() != null && args.getTransmissionDispatcher() != null)
+		{
+			args.getTransmission().incrementNumberOfSends();
+			switch (args.getResponseCode())
+			{
+			case 429:
+			case 439:
+				suspendTransmissions(TransmissionPolicy.BLOCKED_BUT_CAN_BE_PERSISTED, args.getRetryHeader());
+				args.getTransmissionDispatcher().dispatch(args.getTransmission());
+				return true;
+			default:
+				InternalLogger.INSTANCE.trace("Http response code %s not handled by %s", args.getResponseCode(), this.getClass().getName());
+				return false;
+			}        
+		}
+		return false;
+	}
+	
+	
+	private void suspendTransmissions(TransmissionPolicy suspensionPolicy, Header retryAfterHeader) {
+			
+			if (retryAfterHeader == null) {
+				return;
+			}
+			String retryAfterAsString = retryAfterHeader.getValue();
+			if (Strings.isNullOrEmpty(retryAfterAsString)) {
+				return;
+			}
+
+			try {
+				DateFormat formatter = new SimpleDateFormat(RESPONSE_RETRY_AFTER_DATE_FORMAT);
+				Date date = formatter.parse(retryAfterAsString);
+
+				Date now = Calendar.getInstance().getTime();
+				long retryAfterAsSeconds = (date.getTime() - convertToDateToGmt(now).getTime()) / 1000;
+				this.transmissionPolicyManager.suspendInSeconds(suspensionPolicy, retryAfterAsSeconds);
+			} catch (Throwable e) {
+				InternalLogger.INSTANCE.logAlways(InternalLogger.LoggingLevel.ERROR,
+						"Throttled but failed to block transmission, exception: %s", e.getMessage());
+				this.transmissionPolicyManager.backoff();
+			}
+		
+	}
+
+	private static Date convertToDateToGmt(Date date) {
```

Comment:
We need java docs for all the methods and their description. It is crucial for all public methods and private methodsl like this where there is core logic.